### DISCUSSION TITLE
Create many small libs

### DISCUSSION THREAD

**Discussion by dlyr** [STORM-IRIT/Radium-Engine#757] [Design discussion]
As discussed afk, it might be a good idea to split current Radium libs into smaller libs with focused purpose.
There is two questions here:
 - how do we organize code (one directory by lib, or keep as current version with "semantic" organization)
 - how to handle lib dependence, in Radium cmake configuration, but also for client code.

we may use COMPONENTS (as for now) with something like submodule
Core-geom
Core-topology
Core-math
Core-asset
Core-Resources
...

**Comment by dlyr** [comment]
We may start with core, unsing inclusion graph of each cpp file.

**Comment by nmellado** [comment]
Some documentations related to this discussion:
 - Understanding the CMake `COMPONENT` keyword in the `install` command: https://discourse.cmake.org/t/understanding-the-cmake-component-keyword-in-the-install-command/971
 - Doc: link between CPack and CMake `COMPONENT`: https://gitlab.kitware.com/cmake/community/-/wikis/doc/cpack/Component-Install-With-CPack
 - See how cgal handle options, including dependancies management: https://doc.cgal.org/latest/Manual/configurationvariables.html
